src/Polly.Core/ResilienceContext.cs

```diff
@@ -0,0 +1,90 @@
+namespace Polly;
+
+/// <summary>
+/// Context that carries with a single execution through a <see cref="IResilienceStrategy"/>.
+/// </summary>
+/// <remarks>
+/// Do not re-use an instance of <see cref="ResilienceContext"/> across more than one execution.
+/// </remarks>
+public sealed class ResilienceContext
+{
+    internal const bool ContinueOnCapturedContextDefault = false;
+    private ResilienceContext()
+    {
+    }
+
+    /// <summary>
+    /// Gets or sets the <see cref="CancellationToken"/> associated with the execution.
+    /// </summary>
+    public CancellationToken CancellationToken { get; set; }
+
+    /// <summary>
+    /// Gets a value indicating whether the execution is synchronous.
+    /// </summary>
+    public bool IsSynchronous { get; private set; }
```

Comment:
These properties are set by the `Execute` extensions and should not be mutable across the execution.